src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -0,0 +1,219 @@
+import { ExerciseGroup } from 'app/entities/exercise-group.model';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { shortNamePattern } from 'app/shared/constants/input.constants';
+
+@Component({
+    selector: 'jhi-exam-exercise-import',
+    templateUrl: './exam-exercise-import.component.html',
+    styleUrls: ['./exam-exercise-import.component.scss'],
+})
+export class ExamExerciseImportComponent implements OnInit {
+    @Input() exam: Exam;
+    @Input() importInSameCourse = false;
+    // Map to determine, which exercises the user has selected and therefore should be imported alongside an exam
+    selectedExercises = new Map<ExerciseGroup, Set<Exercise>>();
+    // Map with the title and shortName of the programming exercises to be displayed as a placeholder in case of a rejected import
+    titleAndShortNameOfProgrammingExercises = new Map<number, String[]>();
+    // Expose enums to the template
+    exerciseType = ExerciseType;
+    // Map to determine, if an exercise group contains at least one programming exercise.
+    // I.E. the short name must be displayed in the corresponding table
+    containsProgrammingExercises = new Map<ExerciseGroup, boolean>();
+
+    // Patterns
+    // length of < 3 is also accepted in order to provide more accurate validation error messages
+    readonly shortNamePattern = RegExp('(^(?![\\s\\S]))|^[a-zA-Z][a-zA-Z0-9]*$|' + shortNamePattern); // must start with a letter and cannot contain special characters
+    readonly titleNamePattern = RegExp('^[a-zA-Z0-9-_ ]+'); // must only contain alphanumeric characters, or whitespaces, or '_' or '-'
+
+    // Icons
+    faCheckDouble = faCheckDouble;
+    faFileUpload = faFileUpload;
+    faProjectDiagram = faProjectDiagram;
+    faKeyboard = faKeyboard;
+    faFont = faFont;
+
+    constructor() {}
+
+    ngOnInit(): void {
+        this.initializeSelectedExercisesAndContainsProgrammingExercisesMaps();
+        if (this.importInSameCourse) {
+            this.initializeTitleAndShortNameMap();
+        }
+    }
+
+    /**
+     * Method to update the Maps after a rejected import due to invalid project key(s) of programming exercise(s)
+     * Called by the parent component
+     */
+    updateMapsAfterRejectedImport() {
+        this.titleAndShortNameOfProgrammingExercises.clear();
+        this.initializeTitleAndShortNameMap();
+        this.selectedExercises.clear();
+        this.containsProgrammingExercises.clear();
+        this.initializeSelectedExercisesAndContainsProgrammingExercisesMaps();
+    }
+
+    /**
+     * Method to initialize the Maps selectedExercises and containsProgrammingExercises
+     */
+    initializeSelectedExercisesAndContainsProgrammingExercisesMaps() {
+        // Initialize selectedExercises
+        this.exam.exerciseGroups?.forEach((exerciseGroup) => {
+            this.selectedExercises!.set(exerciseGroup, new Set<Exercise>(exerciseGroup.exercises!));
```

Comment:
```suggestion
            this.selectedExercises.set(exerciseGroup, new Set<Exercise>(exerciseGroup.exercises!));
```

`selectedExercises`, `titleAndShortNameOfProgrammingExercises` and `containsProgrammingExercises` cannot be undefined, right? Then you don't need the `!`. (Also for other occurrences)